python/ray/ml/train/integrations/tensorflow/tensorflow_trainer.py

```diff
@@ -0,0 +1,165 @@
+    arguments:
+
+    .. code-block:: python
+
+        def train_loop_per_worker():
+            ...
+
+    .. code-block:: python
+
+        def train_loop_per_worker(config: Dict):
+            ...
+
+    If ``train_loop_per_worker`` accepts an argument, then
+    ``train_loop_config`` will be passed in as the argument. This is useful if you
+    want to tune the values in ``train_loop_config`` as hyperparameters.
+
+    If the ``datasets`` dict contains a training dataset (denoted by
+    the "train" key), then it will be split into multiple dataset
+    shards that can then be accessed by ``ray.train.get_dataset_shard("train")`` inside
+    ``train_loop_per_worker``. All the other datasets will not be split and
+    ``ray.train.get_dataset_shard(...)`` will return the the entire Dataset.
+
+    Inside the ``train_loop_per_worker`` function, you can use any of the
+    :ref:`Ray Train function utils <train-api-func-utils>`.
+
+    .. code-block:: python
+
+        def train_loop_per_worker():
+            # Report intermediate results for callbacks or logging.
+            train.report(...)
+
+            # Checkpoints the provided args as restorable state.
+            train.save_checkpoint(...)
+
+            # Returns dict of last saved checkpoint.
+            train.load_checkpoint()
+
+            # Returns the Ray Dataset shard for the given key.
+            train.get_dataset_shard("my_dataset")
+
+            # Returns the total number of workers executing training.
+            train.get_world_size()
+
+            # Returns the rank of this worker.
+            train.get_world_rank()
+
+            # Returns the rank of the worker on the current node.
+            train.get_local_rank()
+
+    You can also use any of the :ref:`TensorFlow specific function utils
+    <train-api-tensorflow-utils>`.
+
+    .. code-block:: python
+        def train_loop_per_worker():
+            # Turns off autosharding for a dataset.
+            # You should use this if you are doing
+            # `train.get_dataset_shard(...).to_tf(...)`
+            # as the data will be already sharded.
+            train.tensorflow.prepare_dataset_shard(...)
+
+    To save a model to use for the ``TensorflowPredictor``, you must save it under the
+    "model" kwarg in ``train.save_checkpoint()``.
+
+    Example:
+        .. code-block:: python
+
+            import tensorflow as tf
+
+            import ray
+            from ray import train
+            from ray.train.tensorflow import prepare_dataset_shard
+
+            from ray.ml.train.integrations.tensorflow import TensorflowTrainer
+
+            input_size = 1
+
+            def build_model():
+                # toy neural network : 1-layer
+                return tf.keras.Sequential(
+                    [tf.keras.layers.Dense(
+                        1, activation="linear", input_shape=(input_size,))]
+                )
+
+            def train_loop_for_worker():
```

Comment:
Need config here since you are using it for config["num_epochs"]!
```suggestion
            def train_loop_for_worker(config):
```